kernel/cClosure.ml

```diff
@@ -847,6 +847,18 @@ let contract_fix_vect fix =
   in
   (subs_cons(Array.init nfix make_body, env), thisbody)
 
+let unfold_projection info p =
+  if red_projection info.i_flags p
+  then
+    let open Declarations in
+    try
+      let pb = lookup_projection p (info_env info) in
+      Some (Zproj (pb.proj_npars, pb.proj_arg, Projection.constant p))
+    with Not_found ->
+      (* This is possible because sometimes for beta reduction we use a dummy env *)
+      None
```

Comment:
This is messed up. The issue can be witnessed by running some of the test suite files without this exception catching, eg 5761.v (spoilers: we get an uncaught Not_found). 
The issue is at https://github.com/coq/coq/blob/76aff3cbe39da657abb1f559b8ba411a49aab317/pretyping/reductionops.ml#L1244-L1246
```ocaml
let nf_beta = clos_norm_flags CClosure.beta (Global.env ())
let nf_betaiota = clos_norm_flags CClosure.betaiota (Global.env ())
let nf_betaiotazeta = clos_norm_flags CClosure.betaiotazeta (Global.env ())
```
These lines get run at module initialisation time so the environment is always empty.

Instead of catching the exception should we either:
- eta-expand so that the environment is somewhat up to date
- or change the interface so that the caller is in charge of passing the right environment?

Reply:
Second solution (which implies first :) I had a patch to do that but probably before we moved to PRs :/